stellar/deploy-contract.js

```diff
@@ -358,18 +366,39 @@ function sanitizeMigrationData(migrationData) {
     }
 
     if (Array.isArray(parsed)) {
-        return parsed.map(sanitizeMigrationData);
+        return parsed.map((value) => sanitizeMigrationData(value, version, contractName));
+    }
+
+    const custom = customMigrationData(parsed, version, contractName);
+
+    if (custom) {
+        return custom;
     }
 
     if (parsed !== null && typeof parsed === 'object') {
-        return Object.fromEntries(Object.entries(parsed).map(([key, value]) => [key, sanitizeMigrationData(value)]));
+        return Object.fromEntries(Object.entries(parsed).map(([key, value]) => [key, sanitizeMigrationData(value, version, contractName)]));
     }
 
     printInfo('Sanitized migration data', parsed);
 
     return parsed;
 }
 
+function customMigrationData(migrationDataObj, version, contractName) {
+    if (!version) return null;
+    if (!VERSIONED_CUSTOM_MIGRATION_DATA_TYPES[version]) return null;
+    if (!VERSIONED_CUSTOM_MIGRATION_DATA_TYPES[version][contractName]) return null;
```

Comment:
I think you can make it as one liner:
```suggestion
    if (!version || !VERSIONED_CUSTOM_MIGRATION_DATA_TYPES[version]?.[contractName]) return null;
```